library/std/src/sys/pal/unix/fs.rs

```diff
@@ -2113,6 +2113,16 @@ mod remove_dir_impl {
         }
     }
 
+    fn is_enoent(result: &io::Result<()>) -> bool {
+        if let Err(err) = result
+            && matches!(err.raw_os_error(), Some(libc::ENOENT))
+        {
+            true
+        } else {
+            false
+        }
+    }
```

Comment:
```suggestion
    fn ignore_enoent(result: io::Result<()>) -> io::Result<()> {
        if let Err(err) = &result && err.raw_os_error() == Some(libc::ENOENT) {
            Ok(())
        } else {
            result
        }
    }
```

With this function, you can simply wrap the relevant function calls in `ignore_enoent`.

Reply:
> 1. inside the loop, we need to `continue` on ENOENT, not ignore it

As far as I can tell, this actually works with the function. It'd treat removal failed with `ENOENT` the same as successful deletion and continue into the next loop.

> 2\. there are lots of function calls, and almost all of them need to ignore ENOENT. if someone adds a new `unlinkat` call in the future, they will probably also need to wrap it, and if they don't, it will introduce a sneaky regression.

I actually feel the opposite way, it's better if we annotate all the places where we treat `ENOENT` as not an error rather than treating all possible `ENOENT` in this function as non-errors.

E.g. the recursive call does not need to ignore `ENOENT` as far as I can tell, but using try blocks will silence any errors that come from that.

Reply:
> E.g. the recursive call does not need to ignore ENOENT as far as I can tell, but using try blocks will silence any errors that come from that.

actually, maybe we should ignore the error in the recursive call, but not ignore any ENOENT errors outside of the for loop, as top-level ENOENT errors *should* be returned (currently this happens in `remove_dir_all_modern`).